### PR TITLE
add notes regarding etcd 0.4 support and usage example.

### PR DESCRIPTION
**Pull request by kelseyhightower**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,31 @@
...
+
+The current version of go-etcd supports etcd v2.0+, if you need support for both etcd v0.4 and v2.0 please use go-etcd from the [release-0.4](https://github.com/coreos/go-etcd/tree/release-0.4) branch.
+
+```
+package main
+
+import (
+	"log"
+
+    "github.com/coreos/go-etcd/etcd"
```
Comment (xiang90) [inline README.md]: indentation?

File: README.md [hunk 2]
```diff
@@ -1,8 +1,31 @@
...
+[![GoDoc](https://godoc.org/github.com/coreos/go-etcd/etcd?status.png)](https://godoc.org/github.com/coreos/go-etcd/etcd)
+
 The official etcd v0.2 client library for Go.
 
-For usage, please refer to [![GoDoc](https://godoc.org/github.com/coreos/go-etcd/etcd?status.png)](https://godoc.org/github.com/coreos/go-etcd/etcd)
+## Usage
+
+The current version of go-etcd supports etcd v2.0+, if you need support for both etcd v0.4 and v2.0 please use go-etcd from the [release-0.4](https://github.com/coreos/go-etcd/tree/release-0.4) branch.
+
+```
```
Comment (xiang90) [inline README.md]:
```go

File: README.md [hunk 3]
```diff
@@ -1,8 +1,31 @@
 # go-etcd
 
+[![GoDoc](https://godoc.org/github.com/coreos/go-etcd/etcd?status.png)](https://godoc.org/github.com/coreos/go-etcd/etcd)
+
 The official etcd v0.2 client library for Go.
 
-For usage, please refer to [![GoDoc](https://godoc.org/github.com/coreos/go-etcd/etcd?status.png)](https://godoc.org/github.com/coreos/go-etcd/etcd)
+## Usage
+
+The current version of go-etcd supports etcd v2.0+, if you need support for both etcd v0.4 and v2.0 please use go-etcd from the [release-0.4](https://github.com/coreos/go-etcd/tree/release-0.4) branch.
```
Comment (yichengq) [inline README.md]: release-0.4 only supports v0.4
Technically, it uses some deprecated endpoints.

File: README.md [hunk 4]
```diff
@@ -1,8 +1,31 @@
...
+
+```
+package main
+
+import (
+	"log"
```
Comment (yichengq) [inline README.md]: s/\t/    /

File: README.md [hunk 5]
```diff
@@ -1,8 +1,31 @@
 # go-etcd
 
+[![GoDoc](https://godoc.org/github.com/coreos/go-etcd/etcd?status.png)](https://godoc.org/github.com/coreos/go-etcd/etcd)
+
 The official etcd v0.2 client library for Go.
```
Comment (yichengq) [inline README.md]: I guess we need to remove this line.

**Comment by kelseyhightower** [comment]
PTAL

**Comment by xiang90** [comment]
LGTM

**Pull request merged**